.github/workflows/build.yml

```diff
@@ -0,0 +1,83 @@
+name: Build and Deploy
+
+on:
+  push:
+    branches:
+      - staging
+      - master
+  pull_request:
+    branches:
+      - staging
+      - master
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    name: Checks, Tests, & Build
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Repository checkout
+        uses: actions/checkout@v4
+
+      - name: Node.js setup
+        uses: actions/setup-node@v4
+        with:
+          node-version: '16.x'
+
+      - name: 📦 Install dependencies
+        run: yarn install
+
+      - name: ✅ Run Tests
+        run: yarn test --no-watch --coverage
+
+      - name: 🏗️ Build app
+        run: yarn build
+
+      - name: Upload Codecov Reports
+        uses: codecov/codecov-action@v3
+
+  deploy:
+    name: Deploy
+    if: (github.event_name == 'push' || github.event_name == 'workflow_dispatch') && (github.ref == 'refs/heads/master' || github.ref == 'refs/heads/staging')
+    runs-on: ubuntu-latest
+    environment:
+      name: ${{ (github.ref == 'refs/heads/master' && 'production') || 'staging' }}
+    needs: build
+    permissions:
+      id-token: write
+      contents: read
+
+    steps:
+      - name: Repository checkout
+        uses: actions/checkout@v4
+
+      - name: Install AWS CLI
+        run: pip install --user awscli
+
+      - name: Node.js setup
+        uses: actions/setup-node@v4
+        with:
+          node-version: '16.x'
+
+      - name: 📦 Install dependencies
+        run: yarn install --immutable --immutable-cache
+
+      - name: 🏗️ Build app
+        run: yarn build
+        env:
+          REACT_APP_GLOBAL_REGISTRY_BASE_URL: ${{ secrets.GLOBAL_REGISTRY_BASE_URL }}
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-region: us-east-1
+          role-to-assume: ${{ secrets.AWS_GITHUB_ACTIONS_ROLE }}
+
+      - name: Deploy to S3
+        run: |
+          aws s3 sync build/ s3://${{ secrets.AWS_S3_BUCKET }} --acl 'public-read' --cache-control 'max-age=300' --delete
```

Comment:
Unsure if it's needed, but might be worth adding `--region us-east-1`

Reply:
It's present in `give-web` and absent in `know-god-web`, but I can add it.